File: jiggler/jiggler.js
const robot = require("robotjs");
const sin = Math.sin;
const PI = Math.PI;

const screenSize = robot.getScreenSize();
const width = screenSize.width;
const height = screenSize.height / 2 - 10;
let x = 0;

console.log("Por quantos minutos você quer que o mouse se mova?");
process.stdin.on("data", function (data) {
  const minutes = parseInt(data.toString(), 10);
  const endTime = Date.now() + minutes * 60000;

  function moveMouse() {
    if (Date.now() >= endTime) {
      process.exit();
    }

    if (x > width) {
      x = 0;
    }
    for (let i = 0; i < width; i++) {
      const y = height * sin((2 * PI * i) / width) + height;
      robot.moveMouse(i, Math.round(y));
      robot.setMouseDelay(10);
    }
    x++;
  }

  setInterval(moveMouse, 10);
});

File: jiggler/jiggler.kt
import java.awt.Robot
import kotlin.math.sin
import kotlin.math.PI

fun main() {
    val robot = Robot()
    val screenSize = java.awt.Toolkit.getDefaultToolkit().screenSize
    val width = screenSize.width
    val height = screenSize.height / 2 - 10
    var x = 0

    println("Por quantos minutos você quer que o mouse se mova?")
    val minutes = readLine()?.toIntOrNull() ?: 0
    val endTime = System.currentTimeMillis() + minutes * 60_000

    while (System.currentTimeMillis() < endTime) {
        if (x > width) {
            x = 0
        }
        for (i in 0 until width) {
            val y = (height * sin((2 * PI * i) / width) + height).toInt()
            robot.mouseMove(i, y)
            Thread.sleep(10)  // Small delay to slow down the mouse movement
        }
        x++
    }
}
